app/test/process_proposal_test.go

```diff
@@ -234,14 +249,29 @@ func TestProcessProposal(t *testing.T) {
 				btx, _ := coretypes.UnmarshalBlobTx(blobTxs[3])
 				d.Txs = append(d.Txs, btx.Tx)
 			},
+			appVersion:     v1.Version,
 			expectedResult: abci.ResponseProcessProposal_REJECT,
 		},
 		{
-			name:  "undecodable tx",
+			name:  "undecodable tx with app version 1",
 			input: validData(),
 			mutator: func(d *tmproto.Data) {
-				d.Txs = append(d.Txs, tmrand.Bytes(300))
+				d.Txs = append([][]byte{tmrand.Bytes(300)}, d.Txs...)
+				// Update the data hash so that it doesn't fail on an incorrect data root.
+				d.Hash = calculateNewDataHash(t, d.Txs)
+			},
+			appVersion:     v1.Version,
+			expectedResult: abci.ResponseProcessProposal_ACCEPT,
+		},
+		{
+			name:  "undecodable tx with app version 2",
+			input: validData(),
+			mutator: func(d *tmproto.Data) {
+				d.Txs = append([][]byte{tmrand.Bytes(300)}, d.Txs...)
+				// Update the data hash so that it doesn't fail on an incorrect data root.
```

Comment:
[nit] Just for improved clarity:
```suggestion
				// Update the data hash so that the test doesn't fail due to an incorrect data root.
```